src/modules/tasks/global-task-detail/global-task-detail.js

```diff
@@ -0,0 +1,128 @@
+    (function() {
+
+        "use strict";
+
+        angular
+            .module("TendrlModule")
+            .component("globalTaskDetail", {
+
+                restrict: "E",
+                templateUrl: "/modules/tasks/global-task-detail/global-task-detail.html",
+                bindings: {},
+                controller: globalTaskDetailController,
+                controllerAs: "glbTaskDetailCntrl"
+            });
+
+        /*@ngInject*/
+        function globalTaskDetailController($rootScope, $scope, $interval, $state, $stateParams, taskStore, config, utils, clusterStore) {
+
+        var vm = this,
+            statusTimer,
+            msgTimer,
+            isMessagesLoading;
+
+        vm.isDataLoading = true;
+        vm.isMessagesLoading = true;
+        vm.goToClusterTask = goToClusterTask;
+        vm.goToClusterDetail = goToClusterDetail;
+
+        init();
+
+        function _getTaskLogs() {
+            taskStore.getTaskLogs($stateParams.taskId)
+                .then(function(response) {
+                    $interval.cancel(msgTimer);
+
+                    if(typeof vm.taskDetail !== "undefined") {
+
+                        vm.taskDetail.logs = response;
+                        vm.isMessagesLoading = false;
+                    }
+                    startMessageTimer();
+                });
+        }
+
+        function init() {
+
+            $rootScope.selectedClusterOption = "";
+            vm.clusterId = $stateParams.clusterId;
+            if ($rootScope.clusterData) {
+                vm.clusterObj = clusterStore.getClusterDetails(vm.clusterId);
```

Comment:
create a different function for the same set of lines and call a function.

basically, the below lines should in function and call the function from if and else both. will look much cleaner

+                        vm.clusterObj = clusterStore.getClusterDetails(vm.clusterId);
+                        vm.clusterName = vm.clusterObj.cluster_id || "NA";
+                        vm.clusterStatus = clusterStore.checkStatus(vm.clusterObj);
+
+                        taskStore.getJobDetail($stateParams.taskId)
+                            .then(function(data) {
+                                vm.taskDetail = data;
+                                vm.isDataLoading = false;
+
+                                _getTaskLogs();
+                                startStatusTimer();
+                                startMessageTimer();
+                            });